AU2/test/plugins/util/custom_plugins/test_CompetencyPlugin.py

```diff
@@ -75,18 +76,127 @@ def test_competency_plugin_event_create(self):
+        """
+        Test that kills grant auto competency properly.
+        5 players should have competency after this (i.e. the remaining 15 are inco)
+        """
+        p = some_players(20)
+        game = MockGame().having_assassins(p)
+
+        for i in range(5):
+            game.assassin(p[i]).kills(p[i + 5])
+
+        manager = self.get_manager(game, auto_competency=True, initial_competency_period=0)
+        query_date = manager.game_start + datetime.timedelta(days=1, seconds=30)
+
+        incos = manager.get_incos_at(query_date)
+        print(EVENTS_DATABASE.events.values())
+
+        for i in range(5):
+            assert ASSASSINS_DATABASE.get(p[i] + " identifier") not in incos
+
+        for i in range(10):
+            assert ASSASSINS_DATABASE.get(p[i + 10] + " identifier") in incos
+
+        assert len(incos) == 15
+
+    @plugin_test
+    def test_attempts_auto_competency(self):
+        """
+        Test that attempts grant auto competency properly, especially with multiple.
+        Checks whether a player with 0 or 1 attempt is inco, and one with 2, 3, or 2 spread over different events is not
+        """
+        # TODO refactor this to use Alexei's MockGame.add_attempts() once that is merged
+        p = some_players(5)
+        game = (MockGame().having_assassins(p)
+                .assassin(p[1]).is_involved_in_event(pluginState={"CompetencyPlugin": {"attempts": [p[1] + " identifier"]}})
+                .assassin(p[2]).is_involved_in_event(
+                    pluginState={"CompetencyPlugin": {"attempts": [p[2] + " identifier"] * 2}})
+                .assassin(p[3]).is_involved_in_event(
+                    pluginState={"CompetencyPlugin": {"attempts": [p[3] + " identifier"] * 3}})
+                .assassin(p[4]).is_involved_in_event(
+                    pluginState={"CompetencyPlugin": {"attempts": [p[4] + " identifier"]}})
+                .assassin(p[4]).is_involved_in_event(
+                    pluginState={"CompetencyPlugin": {"attempts": [p[4] + " identifier"]}}))
+
+        manager = self.get_manager(game, auto_competency=True, initial_competency_period=0)
+        query_date = manager.game_start + datetime.timedelta(days=1, seconds=30)
+
+        incos = manager.get_incos_at(query_date)
+        print(EVENTS_DATABASE.events.values())
+
+        for i in range(3):
+            assert ASSASSINS_DATABASE.get(p[i+2] + " identifier") not in incos
+
+        for i in range(2):
+            assert ASSASSINS_DATABASE.get(p[i] + " identifier") in incos
+
+        assert len(incos) == 2
+
+    @plugin_test
+    def test_gigabolt(self):
+        """
+        Test that only non-police players with no kills and no attempts will be eliminated.
+        The dangerous logic for gigabolt is in CompetencyPlugin.gigabolt_ask (i.e. the default selections),
+        while this only tests gigabolt_answer directly. It is mostly just code copy-pasted from gigabolt_ask,
+        and thus is not really a good test
```

Comment:
You could perhaps call gigabolt_ask and then extract the list from the `defaults` value of the relevant component?
(sth like `next((c.defaults for c in plugin.gigabolt_ask() if c.identifier == plugin.html_ids["Gigabolt"]))` would do it in one line) then pass this into `gigabolt_answer`?